### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by jboretsky**
This includes the other pull request because I branched from there, but anyhow it has a lot of changes.

The main things I did were:
1) fixed up ngrams classification so it follows the exact same format as jaccard.
2) wrote a combined jaccard/ngram test (using bigrams with jaccard gives ~90% accuracy)
3) separated the feature generations into their own files
4) put all the util functions and generators in a directory called libs/

My idea is that we can actually just have **one** file for the classification, and import all the feature generators. Then we can just comment in/out different features to play around with different combinations.

### CODE REVIEW

File: jaccard_ngrams_combined.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+            num_ngrams_common = 0
+            for ngram in stance_ngrams:
+                if ngram in body_ngrams_dict[stance['Body ID']]:
+                    num_ngrams_common += 1
+            common_ngrams.append(num_ngrams_common)
+
+        return common_ngrams
+
+
+    def _gen_jaccard_sims(self, body_ids, stances):
```
Comment (cpjk) [inline jaccard_ngrams_combined.py]: :fire:

File: jaccard_ngrams_combined.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+
+    def _get_ngrams(self, text, n):
+        tokens = nltk.word_tokenize(text)
+        tokens = [ token.lower() for token in tokens if len(token) > 1 ]
+        ngram_list = list(nltk.ngrams(tokens, n))
+        return ngram_list
+
+
+    def _gen_common_ngrams(self, body_ids, stances, n):
```
Comment (cpjk) [inline jaccard_ngrams_combined.py]: :fire:

File: jaccard_ngrams_combined.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+                num_correct += 1
+        accuracy = num_correct / float(len(predicted))
+        return accuracy
+
+
+    def _process_stance(self, stance):
+        return 'unrelated' if stance == 'unrelated' else 'related'
+
+
+    def _get_ngrams(self, text, n):
```
Comment (cpjk) [inline jaccard_ngrams_combined.py]: :fire:

File: jaccard_ngrams_combined.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+from nltk.classify import NaiveBayesClassifier
+
+from libs.dataset import DataSet
+from libs.gen_ngrams import NgramsGenerator
+from libs.gen_jaccard_sims import JaccardGenerator
+from libs.generate_test_splits import generate_hold_out_split, kfold_split, get_stances_for_folds
+from libs.score import score_submission
+
+
+class JaccardClassify:
```
Comment (cpjk) [inline jaccard_ngrams_combined.py]: change class name?

Comment (jboretsky) [reply jaccard_ngrams_combined.py]: roight mb

Comment (cpjk) [reply jaccard_ngrams_combined.py]: :australia:

**Comment by cpjk** [comment]
thumbs_up

**Pull request merged**